docs/sdk-configuration.md

```diff
@@ -0,0 +1,492 @@
+reference to "the SDK" in this document refers to the full SDK with all signals.
+
+- Make sure everything is auto-configurable. This is out of the scope of the SDK, and instead is
+left to auto-configuration layers, which are also described below but not as part of the core SDK. 
+In partiular, SignalProvider SPIs which currently exist are proposed to be removed from the SDK. The 
+SDK may provide an autoconfiguration extension as an option which is not internal to the main SDK 
+components.
+
+## Configuring an instance of the SDK
+
+The SDK consists of configuration options for all the signals it supports. Users all have different
+requirements for how they use the SDK, for example they may use different exporters depending on
+their backend. Because we cannot guess the configuration the user needs, we expect that the SDK must
+be configured by the user before it can be used.
+
+Goals for configuring the SDK are
+
+- Discoverability of options
+- Ease of use by end users, e.g., less complicated code required
+- Avoid requiring duplicate configuration, which can lead to errors or confusion
+- Provide good defaults where possible
+
+In Java, the builder pattern is common for configuring instances. Let's look at what that may look
+like. The simplest configuration will be when a user wants to get a default experience, exporting
+with a specific exporter to an endpoint.
+
+```java
+OpenTelemetrySdk openTelemetry = OpenTelemetrySdk.builder()
+  .addSpanExporter(OtlpGrpcSpanExporter.builder()
+    .setEndpoint("spans-service:4317")
+    .build())
+  .addMetricExposition(PrometheusExposition.getDefault())
+  .build();
+```
+
+This
+
+- Exports spans using OTLP to spans-service
+  - Uses the BatchSpanProcessor (or a batching exporter wraper) which we consider the best default 
+  for exporting. It has defaults for queue size, timeout, etc
+- Exposes metrics using Prometheus format
+- Uses ParentBased(AlwaysOn) sampler
+- Uses standard random IDs
+- Uses a default Resource which simply consists of the SDK (or any other resources we decide to)
+include in the core SDK, not extensions
+- Uses the default Clock, which uses Java 8 / 9+ optimized APIs for getting time
+  - The only real reason a user would set this is for unit tests, not for production
+- Enforces default numeric limits related to number of attribute, etc
+
+Because the exporting is often the only aspect an end user needs to configure, this is the simplest
+possible API for configuring the SDK.
+
+Let's look at a more complicated example
+
+```java
+OpenTelemetrySdk openTelemetry = OpenTelemetrySdk.builder()
+  .setResource(Resource.getDefault().merge(CoolResource.getDefault()))
+  .setClock(AtomicClock.create())
+  .setPropagators(B3Propagator.getInstance())
+  .addSpanExporter(OtlpGrpcSpanExporter.builder()
+    .setEndpoint("spans-service:4317")
+    .setTimeout(Duration.ofSeconds(10))
+    .setBatchQueueSize(10000)
```

Comment:
batching becomes the property of exporters?

Reply:
It's a random idea - more that batching configuration is the job of exporter configuration. One goal was to make it more obvious when an exporter is tied to a batching (or not) strategy.

But as with the comment about async processor this might be too limiting so not liking it so much anymore but still want to talk it through a bit.